### PR TITLE
Use version catalog in buildSrc code

### PR DESCRIPTION
**Pull request by PawelLipski**

### CODE REVIEW

File: buildSrc/build.gradle.kts [hunk 1]
```diff
@@ -1,7 +1,6 @@
 import org.jetbrains.kotlin.gradle.tasks.KotlinCompile
 
 plugins {
-  `java-library`
```
Comment (mkondratek) [inline buildSrc/build.gradle.kts:4]: image

File: buildSrc/settings.gradle.kts [hunk 2]
```diff
@@ -1,3 +1,8 @@
+
 dependencyResolutionManagement {
-  versionCatalogs { create("libs") { from(files("../gradle/libs.versions.toml")) } }
+  versionCatalogs {
+    create("libs") {
```
Comment (mkondratek) [inline buildSrc/settings.gradle.kts:4]: so we were not using it b4 although the catalog was present, right?

File: buildSrc/src/main/kotlin/com/virtuslab/gitmachete/buildsrc/BuildUtils.kt [hunk 3]
```diff
@@ -92,70 +106,71 @@ object BuildUtils {
 
     project.dependencies {
       add("uiTestImplementation", testFixtures(project(":testCommon")))
-      add("uiTestImplementation", "org.virtuslab.ideprobe:junit-driver_2.13:0.36.1")
-      add("uiTestImplementation", "org.virtuslab.ideprobe:robot-driver_2.13:0.36.1")
+      add("uiTestImplementation", project.bundle("ideProbe"))
```
Comment (mkondratek) [inline buildSrc/src/main/kotlin/com/virtuslab/gitmachete/buildsrc/BuildUtils.kt:109]: nice. finally it look like człowiek

File: buildSrc/src/main/kotlin/com/virtuslab/gitmachete/buildsrc/CheckerFrameworkConfigurator.kt [hunk 4]
```diff
@@ -1,13 +1,14 @@
...
+import com.virtuslab.gitmachete.buildsrc.BuildUtils.lib
 import org.checkerframework.gradle.plugin.CheckerFrameworkExtension
 import org.checkerframework.gradle.plugin.CheckerFrameworkPlugin
 import org.gradle.api.Project
 import org.gradle.api.plugins.JavaPluginConvention
 import org.gradle.api.tasks.SourceSet
 import org.gradle.kotlin.dsl.*
 
-//  TODO (#1004): Remove hardcoded dependencies and deprecated code
+//  TODO (#1004): Remove deprecated code
```
Comment (mkondratek) [inline buildSrc/src/main/kotlin/com/virtuslab/gitmachete/buildsrc/CheckerFrameworkConfigurator.kt:11]: one check box resolved 😌

**Review by mkondratek — APPROVED** [review]

File: buildSrc/settings.gradle.kts (hunk 2 above)
Comment (PawelLipski) [reply buildSrc/settings.gradle.kts:4]: Hmm there was no `buildSrc` until recently!

File: buildSrc/build.gradle.kts (hunk 1 above)
Comment (PawelLipski) [reply buildSrc/build.gradle.kts:4]: Was redundant!

File: buildSrc/settings.gradle.kts (hunk 2 above)
Comment (PawelLipski) [reply buildSrc/settings.gradle.kts:4]: Yes, exactly, there were only hardcoded references to libs in buildSrc kotlin code

**Review by MaciejG604 — APPROVED** [review]

**Pull request merged**